src/include/cursor.i

```diff
@@ -17,6 +17,69 @@ __cursor_set_recno(WT_CURSOR_BTREE *cbt, uint64_t v)
     cbt->iface.recno = cbt->recno = v;
 }
 
+/*
+ * __cursor_copy_release_item --
+ *     Release memory used by the key or value item in cursor copy debug mode.
+ */
+static inline int
+__cursor_copy_release_item(WT_SESSION_IMPL *session, WT_ITEM *item)
+{
+    WT_DECL_ITEM(tmp);
+    WT_DECL_RET;
+    void *extra_mem;
+
+    if (!F_ISSET(S2C(session), WT_CONN_DEBUG_CURSOR_COPY) || item->mem == NULL)
+        return (0);
+
+    /*
+     * If we own the memory for the item, make a copy and use that instead. That allows us to
+     * overwrite and free the original memory, potentially uncovering programming errors related to
+     * retaining pointers to key/value memory beyond API boundaries. If the item has memory not
+     * currently in use, free that as well.
+     */
+    if (WT_DATA_IN_ITEM(item)) {
+        /*
+         * We need an extra allocation and a free of it at the right time. Otherwise, with some
+         * allocators, we may get the exact same memory address back in the item. and we will never
+         * catch any accesses to the original memory. Ideally, We want the memory originally used by
+         * the item to end up on the malloc free list when we return.
+         */
+        WT_ERR(__wt_malloc(session, item->size, &extra_mem));
+        WT_ERR(__wt_scr_alloc(session, 0, &tmp));
+        WT_ERR(__wt_buf_set(session, tmp, item->data, item->size));
+        __wt_explicit_overwrite(item->mem, item->memsize);
+        __wt_buf_free(session, item);
+        __wt_free(session, extra_mem);
+        WT_ERR(__wt_buf_set(session, item, tmp->data, tmp->size));
+    } else {
+        __wt_explicit_overwrite(item->mem, item->memsize);
+        __wt_buf_free(session, item);
+    }
+err:
+    __wt_scr_free(session, &tmp);
+    return (ret);
+}
+
+/*
+ * __cursor_copy_release_key --
+ *     Release memory used by the key in cursor copy debug mode.
+ */
+static inline int
+__cursor_copy_release_key(WT_SESSION_IMPL *session, WT_CURSOR *cursor)
+{
+    return (__cursor_copy_release_item(session, &cursor->key));
+}
+
+/*
+ * __cursor_copy_release_value --
+ *     Release memory used by the value in cursor copy debug mode.
+ */
+static inline int
+__cursor_copy_release_value(WT_SESSION_IMPL *session, WT_CURSOR *cursor)
+{
+    return (__cursor_copy_release_item(session, &cursor->value));
+}
```

Comment:
These two functions could just take a `WT_CURSOR` and we can get the `WT_SESSION` from it (that's what most of the other cursor functions do, I think, and it saves you work in `__cursor_localkey()` and `__cursor_localvalue()`.

Reply:
Fixed.